lib/invoice_repo.rb

```diff
@@ -20,4 +21,40 @@ def find_all_by_merchant_id(id)
   def find_all_by_status(status)
     @repository.select { |invoice| invoice.status == status }
   end
+
+  def invoice_status(type)
+    count = find_all_by_status(type).count
+    total = all.count
+    percent(count, total)
+  end
+
+  def number_of_invoices_per_day
+    days = Hash.new(0)
```

Comment:
I'm not sure if days is used anywhere else but if it is we can consider storing this hash like we do with a merchants items.